timetable/utils.py

```diff
@@ -48,13 +48,19 @@ def get_timetable_context(lessons):
 
     # TODO: a cleaner way to pass period str to the template while using
     #       period number as key?
+    alternative_periods = Period.objects.exclude(schedule=1) # Exclude default periods
+    alternative_day_plan = None
+    for day_plan in DayPlan.objects.all():
+        if day_plan.is_today:
+            alternative_day_plan = day_plan
+            break
```

Comment:
In most cases, if you have to use `break` in your code, something is poorly designed. Why are you iterating over all the DayPlans if you can just ask the database for a DayPlan for today in try-except block?

Reply:
I have simplified the query on DayPlan table as well, thanks for the suggestion!